### PR TITLE
Hide str flag if unchanged and string shown + asm.noisy

### PR DESCRIPTION
**Pull request by kazarmy**

### CODE REVIEW

File: libr/core/disasm.c [hunk 1]
```diff
@@ -2600,6 +2600,27 @@ static void ds_print_asmop_payload(RDisasmState *ds, const ut8 *buf) {
 	}
 }
 
+static inline bool is_filtered_flag(RDisasmState *ds, const char *name) {
+	if (strncmp (name, "str.", 4)) {
+		return false;
+	} else {
```
Comment (radare) [inline libr/core/disasm.c]: dont else after return, avoid unnecessary  nesting

**Comment by radare** [comment]
more readable, but hiding the flags so i would like to discuss before merging this.

screen shot 2017-05-20 at 13 36 24

**Comment by kazarmy** [comment]
Any decision on this?

**Comment by radare** [comment]
i think we shuold add an asm.noisy var for this, as well and other comments that doesnt make much sense, but could be useful in some situations. like those ones: ; [0x30:4]=-1

**Comment by kazarmy** [comment]
Can I name it asm.allcmts? Easier to make it more granular if needed e.g.
asm.allflagcmts.

**Comment by radare** [comment]
the signature comments are not aligned, can u fix that too?

**Comment by kazarmy** [comment]
If you mean zignature comments, then will have to do the asm.noisy var first but it should be technically possible. RL will intervene for the next few days (and nights) though.

**Comment by radare** [comment]
lets name it noisy for now and we can discuss the name meanwhile
thanks!

**Comment by radare** [comment]
in fact this flag should appear in the lea [rip + xxx] but for some reason relsub is not working here. can you have a look at this and rebase the branch?

thanks!

**Comment by kazarmy** [comment]
The test sets `asm.relsub` to false. Setting it to true works fine.

**Comment by kazarmy** [comment]
asm.noisy currently defaults to false. The status quo is kept if asm.noisy is true. Should I default it to true?

**Comment by Maijin** [comment]
well if asm.relsub show the flag in the disassembly there is no reason to keep showing it on the right  not even sure of the need of the variable.

**Comment by radare** [comment]
I think that having this flag set to true by default will allow us to continue experimenting yhis without breaking master or changing behaviours and remove it when ready. But yep that relsub stuff is unrelated to this pr but should be done too

**Comment by kazarmy** [comment]
> I think that having this flag set to true by default ...

Okay. The zignature comment fix should be in a different pr. Can merge now?

**Comment by kazarmy** [comment]
> well if asm.relsub show the flag in the disassembly there is no reason to keep showing it on the right ...

Currently it isn't shown. Need a test case if it is.

**Pull request merged**